### PR TITLE
ENG-34175: support endpoint resolution for websockets

### PR DESCRIPTION
**Pull request by BryanTabarez**
* Support endpoint resolution for websockets
* Add 'responder' to resolveByResidencyServiceList

### CODE REVIEW

File: src/client/al-api-client.ts [hunk 1]
```diff
@@ -718,7 +718,11 @@ export class AlApiClient implements AlValidationSchemaProvider
       let response = await this.axiosRequest( endpointsRequest );
       Object.entries( response.data ).forEach( ( [ serviceName, endpointHost ] ) => {
           let host = endpointHost as string;
-          host = host.startsWith("http") ? host : `https://${host}`;      //  ensuring domains are prefixed with protocol
+          if (host.includes("async") && !host.startsWith("http") && !host.startsWith("ws")) {
+            host = `wss://${host}`; // add prefix for websocket protocol
+          } else {
+            host = host.startsWith("http") ? host : `https://${host}`;      //  ensuring domains are prefixed with protocol
+          }
```
Comment (parky128) [inline src/client/al-api-client.ts:725]:
```suggestion
          if (host.startWith("async.")) {
            host = `wss://${host}`; // add prefix for websocket protocol
          } else {
            host = host.startsWith("http") ? host : `https://${host}`;      //  ensuring domains are prefixed with protocol
          }
```

File: src/client/al-api-client.ts [hunk 2]
```diff
@@ -844,7 +848,12 @@ export class AlApiClient implements AlValidationSchemaProvider
           Object.entries(residencyLocations).forEach(([residencyName, residencyHost]) => {
               Object.entries(residencyHost).forEach(([datacenterId, endpointHost]) => {
                 let host = endpointHost as string;
-                host = host.startsWith("http") ? host : `https://${host}`;      //  ensuring domains are prefixed with protocol
+                if (host.includes("async") && !host.startsWith("http") && !host.startsWith("ws")) {
+                  host = `wss://${host}`; // add prefix for websocket protocol
+                  console.warn("host", host);
+                } else {
+                  host = host.startsWith("http") ? host : `https://${host}`;      //  ensuring domains are prefixed with protocol
+                }
```
Comment (parky128) [inline src/client/al-api-client.ts:856]: Same suggestion here as earlier comment above

File: src/client/al-api-client.ts [hunk 3]
```diff
@@ -718,7 +718,11 @@ export class AlApiClient implements AlValidationSchemaProvider
       let response = await this.axiosRequest( endpointsRequest );
       Object.entries( response.data ).forEach( ( [ serviceName, endpointHost ] ) => {
           let host = endpointHost as string;
-          host = host.startsWith("http") ? host : `https://${host}`;      //  ensuring domains are prefixed with protocol
+          if (host.startsWith("async")) { // naming convention for WebSocket services
+            host = `wss://${host}`; // add prefix for websocket protocol
+          } else {
+            host = host.startsWith("http") ? host : `https://${host}`;      //  ensuring domains are prefixed with protocol
```
Comment (mcnielsen) [inline src/client/al-api-client.ts:724]: This is actually an `else if` disguised as an `else` + ternary.  Avoiding unnecessary reassignment can avoid accidents, so I'd prefer

```
else if ( host.startsWith( "http" ) {
    host = `https://${host}`;
}

File: src/client/al-api-client.ts [hunk 4]
```diff
@@ -844,7 +848,12 @@ export class AlApiClient implements AlValidationSchemaProvider
           Object.entries(residencyLocations).forEach(([residencyName, residencyHost]) => {
               Object.entries(residencyHost).forEach(([datacenterId, endpointHost]) => {
                 let host = endpointHost as string;
-                host = host.startsWith("http") ? host : `https://${host}`;      //  ensuring domains are prefixed with protocol
+                if (host.startsWith("async")) { // naming convention for WebSocket services
+                  host = `wss://${host}`; // add prefix for websocket protocol
+                  console.warn("host", host);
+                } else {
+                  host = host.startsWith("http") ? host : `https://${host}`;      //  ensuring domains are prefixed with protocol
```
Comment (mcnielsen) [inline src/client/al-api-client.ts:855]: and again here.

**Review by mcnielsen — APPROVED** [review]

**Pull request merged**